lib/parser.js

```diff
@@ -955,42 +956,51 @@ module.exports = function parse (line) {
 
     // very new variable
     if (keys[0] === 'very') {
-        statement += 'var ' + keys[1] + ' = ';
-        if (keys[3] === 'new') {
-          // shift tokens
-          var newCallKeys = keys.slice(3);
-          statement += handleNew(newCallKeys);
+      // consume: very
+      keys.shift();
+
+      statement += 'var ' + keys.shift() + ' = ';
+
+      // consume:is/as
+      keys.shift();
+
+      if (keys[0] === 'new') {
+        statement += handleNew(keys);
+        return statement;
+      }
+
+      if (keys[0] === 'much') {
+        var anonStatement = handleLambda(keys);
+        statement += anonStatement;
+        return statement;
+      }
+
+      if (keys[0] === 'obj') {
+        statement += '{\n';
+        multiLine = true;
+        return statement;
+      }
+
+      if (keys.length > 1) {
+        if (isDogescriptSource(keys)) {
+          statement += parse(joinTokens(keys));
           return statement;
         }
-        if (keys[3] === 'much') {
-            // shift tokens
-            var anonKeys = keys.slice(3);
-            var anonStatement = handleLambda(anonKeys);
-            statement += anonStatement;
-            return statement;
-        }
-        if (keys.length > 4) {
-            var recurse = '';
-            for (var i = 3; i < keys.length; i++) {
-                if (keys[i].substr(-1) === ',' && keys[i].charAt(keys[i].length - 2) !== '}') keys[i] = keys[i].slice(0, -1);
```

Comment:
not entirely sure what this check was about....... probably a copy from how plz would add `,` to each token so splitting `{ a: b }` would break as `{`, `a:`, `b`, `}` . Couldn't think of a scenario for very that would trigger this tho